javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,6 +56,24 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
+        if (size == 0) { return empty(); }
+
+        Object[] array = (iterable instanceof Collection<?>) ? ((Collection<?>) iterable).toArray()
+                                                             : asArray(iterable.iterator(), size);
+        return ofAll(array, size, 0);
```

Comment:
Style-alert :) `else` branch missing (two related disjunct cases) and `final` keyword at `Object[] array`